### ISSUE TITLE
Feature Request: Add utility methods for creating flow runs

### ISSUE THREAD

**Issue by jacksund** [PrefectHQ/prefect#4616]
## Current behavior

- Flow runs are created via `client.create_flow_run` method, CLI, or UI. One can also do a local run with `flow.run` and even start a Agent for a given flow with `flow.run_agent`
- The creation of each flow run must also make a separate call to the server.

## Proposed behavior

- a new method that creates/registers flow runs on Prefect Server/Cloud is attached directly to flow objects (i.e. `flow.run_client`)
- If a large number of flow runs with varying parameters needs to be made, there is a `client.create_flow_runs` method that creates these runs with a single call to the server. Note this is a bulk write to the database and is analogous to a mapping method across a list of parameter dictionaries. Likewise a `flow.run_many_client` method could also be useful. 

## Examples

Example 1
```python
# after creating a flow I can register it with Prefect cloud
flow.register(project="Example-Project")

# Once a flow is registered, I can also create a new flow run on Prefect cloud too.
# This is really just a convenience wrapper for client.create_flow_run
flow.run_cloud()
```

Example 2
```python
# this is extremely useful for module-based flows where you import
# the flow and create new flow runs dynamically in scripts
from mypackage.flows import ExampleFlow as myflow
myflow.run_cloud()

# with the current API, this would be a bit more boilerplate
from prefect import Client
from mypackage.flows import ExampleFlow as myflow
client = Client()
client.create_flow_run(name=myflow.name)
```

Example 3
```python
# let's say we have a flow that we want to run many times with various parameters
parameters_mapped = [{...}, {...}, {...}, ...]  

# With the current API, submitting many flows is very slow and requires a for-loop
for parameters in parameters_mapped:
    client.create_flow_run(name=..., parameters=parameters)

# A single bulk query could make this much easier on the server and must faster overall
client.create_flow_runs(parameters=parameters_mapped)
```

**Comment by madkinsz** [comment]
Thanks for the issue @jacksund --

`flow.cloud_run` is on my roadmap and actually was the impetus for #4434 but has some complications as a flow must be registered to be run and we do not want to implicitly register a flow yet. Then, once the flow is registered we must store the id it has been assigned in the flow object which does not persist between script loads. We're trying to think of the best way to make this a smooth process. 

One thing on my roadmap is tearing API calls out of the Client which would turn

```python
from prefect import Client
from mypackage.flows import ExampleFlow as myflow
client = Client()
client.create_flow_run(name=myflow.name)
```

into

```python
from mypackage.flows import ExampleFlow as myflow
from prefect.backend import create_flow_run

create_flow_run(register_flow(myflow))
```


Another thing I'm imaging which would address your second example is something like this with the CLI

```bash
prefect register -m mypackage.flows.ExampleFlow --quiet | prefect run --id
```

Your third example would require us to introduce a new route in the backend, which we can certainly consider, but generally we want our routes to do _one_ thing. For example, while creating 5 flow runs, what should we do if one of them fails to be created? How do we return that information to the user in a meaningful way? etc.

**Comment by jacksund** [comment]
Thanks for the reply!

> once the flow is registered we must store the id it has been assigned in the flow object which does not persist between script loads

I am a little lost on this. Instead of storing a specific flow-id, are you able to call `create_flow_run(flow_name=self.name)` and assume the newest version? I'm looking at your developmental method here but haven't tested it out yet: https://github.com/PrefectHQ/prefect/blob/7260c3482923a6440dd46360213c4481c7b5b455/src/prefect/tasks/prefect/flow_run.py#L63

I can see an issue when there are flows with the same name in multiple projects though. In that case, maybe introducing a `flow.project_name` property to the flow class would do the trick...? Idk I'm spitballing.

And thanks for the 2nd example! I'll definitely give this a go.

> while creating 5 flow runs, what should we do if one of them fails to be created? How do we return that information to the user in a meaningful way?

Yeah, this is a good point. I commonly use django's [bulk_create and bulk_update](https://docs.djangoproject.com/en/3.0/ref/models/querysets/#bulk-create) methods, where these are always treated as atomic transactions (so one failure means all modifications are rolled back). Errors are always raised in python because of the model validation -- and I'd guess Prefect does something similar with [marshmallow validation](https://marshmallow.readthedocs.io/en/stable/examples.html#validating-package-json). 

The overhead from the independent database hits/mutations isn't a huge factor for me now, but I'd definitely be interested in this down the road.

**Comment by madkinsz** [comment]
> I can see an issue when there are flows with the same name in multiple projects though. In that case, maybe introducing a flow.project_name property to the flow class would do the trick...? Idk I'm spitballing.

Yeah this is precisely the issue with using the name since they're not guaranteed to be unique. We could store the project name on the flow as well and we could allow flows to exist without a project and default to a null project (#4378), but this work is a bit further out yet.

Generally, we're looking to make it easier to dynamically create and execute flow runs. You're very likely to see the improvements you're looking for, but I can't provide a timeline yet.

> I commonly use django's bulk_create and bulk_update methods, where these are always treated as atomic transactions (so one failure means all modifications are rolled back)

cc @cicdw and @zangell44 regarding bulk flow run creation routes. We'll consider this as the scale of API usage grows.
